overlord/servicestate/quota_handlers.go

```diff
@@ -823,7 +854,75 @@ func validateSnapForAddingToGroup(st *state.State, snaps []string, group string,
 			}
 		}
 	}
+	return nil
+}
 
+// splitSnapServiceName splits and verifies the snap service reference
+// taken in by the frontend. It expects the format snap.service
+func splitSnapServiceName(name string) (string, string, error) {
+	parts := strings.Split(name, ".")
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("invalid snap service: %s", name)
+	}
+	if err := naming.ValidateSnap(parts[0]); err != nil {
+		return "", "", err
+	}
+	return parts[0], parts[1], naming.ValidateApp(parts[1])
+}
+
+// appReferenceIsService returns whether the service referred to in the
+// snap is actually a service.
+func appReferenceIsService(st *state.State, snap, service string) error {
```

Comment:
If you really want to return an error, this should be called `ensureAppReferenceIsService` (or s/ensure/check/? Samuele might have another opinion :-) ); otherwise keep the current name and return a boolean.

Reply:
Ah thank you, this is a great point